src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -0,0 +1,159 @@
+  Formik,
+  FormikForm as Form,
+  FormInput,
+  Heading,
+  Layout,
+  ModalErrorHandler,
+  ModalErrorHandlerBinding,
+  PageSpinner,
+  SelectOption
+} from '@wings-software/uicore'
+import { FontVariation } from '@harness/design-system'
+import * as Yup from 'yup'
+import { useParams } from 'react-router-dom'
+import { useStrings } from 'framework/strings'
+import { useGetFolder, ErrorResponse } from 'services/custom-dashboards'
+import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
+import css from './HomePage.module.scss'
+
+const TAGS_SEPARATOR = ','
+
+interface DashboardFormProps {
+  formData?: DashboardFormRequestProps
+  title: string
+  loading: boolean
+  onComplete: (data: DashboardFormRequestProps) => void
+  modalErrorHandler?: ModalErrorHandlerBinding
+  setModalErrorHandler: (modalErrorHandler: ModalErrorHandlerBinding) => void
+}
+
+export interface DashboardFormRequestProps {
+  description?: string
+  folderId: string
+  name: string
+}
+
+export interface DashboardFormikValues {
+  description: string[]
+  folderId: string
+  name: string
+}
+
+const DashboardForm: React.FC<DashboardFormProps> = ({
+  formData,
+  title,
+  loading,
+  onComplete,
+  modalErrorHandler,
+  setModalErrorHandler
+}) => {
+  const { getString } = useStrings()
+  const { accountId } = useParams<AccountPathProps>()
+  const [initialValues, setInitialValues] = React.useState<DashboardFormikValues>({
+    folderId: formData?.folderId || '',
+    name: formData?.name || '',
+    description: formData?.description?.split(TAGS_SEPARATOR) || []
+  })
+  const sharedFolder = {
+    value: getString('dashboards.shared'),
+    label: getString('dashboards.sharedFolderTitle')
+  }
+  const [folderListItems, setFolderListItems] = React.useState<SelectOption[]>([sharedFolder])
+
+  const {
+    data: foldersList,
+    error,
+    loading: isFoldersLoading
+  } = useGetFolder({
+    queryParams: { accountId: accountId }
+  })
+
+  function isMissingFolderId(folders: SelectOption[]): boolean {
+    return !folders.some((item: SelectOption) => item.value === formData?.folderId)
+  }
+
+  React.useEffect(() => {
+    const errorResponse = error?.data as ErrorResponse
+    if (errorResponse?.responseMessages) {
+      modalErrorHandler?.showDanger(errorResponse.responseMessages)
+    }
+  }, [error, modalErrorHandler])
+
+  React.useEffect(() => {
+    if (foldersList?.resource) {
+      const folders = foldersList?.resource?.map((folder: { id: string; name: string }) => {
+        return { value: folder?.id, label: folder?.name }
+      })
+      if (isMissingFolderId(folders)) {
+        setInitialValues(prevState => {
+          return { ...prevState, folderId: 'shared' }
+        })
+      }
+      setFolderListItems(prevState => [...prevState, ...folders])
+    }
+  }, [foldersList])
+
+  return (
+    <Layout.Vertical padding="xxlarge">
+      <Heading level={3} font={{ variation: FontVariation.H3 }} padding={{ bottom: 'large' }}>
```

Comment:
Another totally optional one, but since the `padding` is to keep the title and form separate, you can just use the `spacing` prop on the `Layout` above.
```suggestion
    <Layout.Vertical padding="xxlarge" spacing="large">
      <Heading level={3} font={{ variation: FontVariation.H3 }}>
```